libs/payments/metrics/src/lib/glean/glean.provider.ts

```diff
@@ -13,12 +15,19 @@ export const PaymentsGleanProvider = Symbol('GleanServerEventsProvider');
 export const PaymentsGleanFactory: Provider<PaymentsGleanServerEventsLogger> = {
   provide: PaymentsGleanProvider,
   useFactory: (config: PaymentsGleanConfig) => {
+    // In development, we want to have a unique logger name. Without it, each time the
+    // nestapp restarts, a new logger handler is added and  the events will be logged
+    // multiple times, once per handler.
+    const loggerAppName =
+      process.env['NODE_ENV'] === 'development'
+        ? `${config.loggerAppName}-${getRandomValue()}`
```

Comment:
Is the preference for this over something like uuidv4() just for line lengths? Seems like it might be a more semantic way to do this from the code side of things

Reply:
Mostly, I didn't want to add an additional dependency. Since this is just for dev, and it's not super critical that it's always a unique value, I figured a little random functions is fine.

Reply:
Not a tweak or anything but more of a fun fact -- crypto.randomUUID is built into node:

https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID

Reply:
TIL thank you!